lib/dao.js

```diff
@@ -1683,7 +1683,27 @@ DataAccessObject._coerce = function(where, options) {
     }
     var DataType = props[p] && props[p].type;
     if (!DataType) {
-      continue;
+      // Handle nested properties
+      if (p.indexOf('.') !== -1) {
+        var parts = p.split('.');
+        var end = parts.pop();
+        var obj = props;
+        while (parts.length) {
+          var part = parts.shift();
+          if (typeof obj[part].type !== 'function' ||
+            !obj[part].type.definition ||
+            !obj[part].type.definition.properties) {
```

Comment:
We are trying to avoid multi-line `if` conditions, see http://loopback.io/doc/en/contrib/style-guide.html#indentation-of-multi-line-expressions-in-if